plugins/catalog-graph/src/components/EntityRelationsGraph/useEntityRelationGraph.ts

```diff
@@ -81,12 +88,19 @@ export function useEntityRelationGraph({
 
       ++depth;
     }
-
     requestEntities([...expectedEntities]);
-  }, [entities, rootEntityRefs, maxDepth, relations, kinds, requestEntities]);
+  }, [
+    entities,
+    rootEntityRefs,
+    maxDepth,
+    relations,
+    kinds,
+    func,
+    requestEntities,
+  ]);
 
   return {
-    entities,
+    entities: pickBy(entities, (value, _key) => !func || func(value)),
```

Comment:
would do something like this `entities: func ? entities.filter(func) : entities,`.
This way we check once if `func` exists and avoid an compete interation if not.